### DISCUSSION TITLE
Linearity of Binary Logistic Regression

### DISCUSSION THREAD

**Discussion by Ajith-Shenoy** [Lightning-AI/dl-fundamentals#61] [Lecture videos and quizzes] [answered]
Is a single layer logistic neuron with sigmoid activation able to learn non linear classification boundaries ?
There is a little bit of ambiguity when it comes to this in the lectures at places.
1. Logistic regression is better than perceptron as it always converges.
2.  It can learn non linear boundaries.
I looked it up online and found that it can only still capture linear decision boundaries. Why is it so when sigmoid is non-linear activation function ?
Is it non-linear for binary classification only ? Please clarify. Thanks

**Comment by rasbt** [comment, marked as answer]
> Is a single layer logistic neuron with sigmoid activation able to learn non linear classification boundaries ?

Good question, but the answer is no. This would be a logistic regression model :).

> Why is it so when sigmoid is non-linear activation function ?

That's because the terms still enter the function in a linear fashion. E.g., if you have `sigmoid(w1*x1 + w2*x2 + b)` then `w1*x1 + w2*x2 + b` is still a linear function. To create non-linear boundaries, there would need to be a nonlinear interaction between the terms. E.g., `w1*x1 + w2*x2^2 + b`  or `w1*x1 + w2*w1*x2 + b` etc.
